### PR TITLE
Disable single dispatch in EH extension

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/Azure/azure-sdk-for-net/issues/19874

We are going to bring it back so don't rip the code out.

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Triggers/EventHubTriggerAttributeBindingProvider.cs [hunk 1]
```diff
@@ -54,9 +54,15 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
             Func<ListenerFactoryContext, bool, Task<IListener>> createListener =
              (factoryContext, singleDispatch) =>
              {
+                 var options = _options.Value;
+                 if (singleDispatch && !options.IsSingleDispatchEnabled)
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Triggers/EventHubTriggerAttributeBindingProvider.cs:58]: Why do this? We already have a test for single dispatch so this code will be hit.

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Triggers/EventHubTriggerAttributeBindingProvider.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Triggers/EventHubTriggerAttributeBindingProvider.cs:58]: or is it still possible to bind to this via config even though it is now internal?

**Comment by pakrym** [comment]
cc @paulbatum

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Triggers/EventHubTriggerAttributeBindingProvider.cs [hunk 2]
```diff
@@ -54,9 +54,15 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
...
+                 {
+                     throw new NotSupportedException("The single dispatch is not supported, please use batch processing (switch the bound parameter to an array).");
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Triggers/EventHubTriggerAttributeBindingProvider.cs:60]:
```suggestion
                     throw new NotSupportedException("Binding to individual events is not supported. Please use batch processing by binding to an array instead.");
```

**Pull request merged**